src/components/Table/DataTable.stories.tsx

```diff
@@ -80,3 +81,95 @@ export const DefaultVariant: Story = {
 		data: data
 	}
 };
+
+export const CustomizedVariant: Story = {
+	name: "Customized",
+	render: () => <CustomizedDataTable />,
+	args: {
+		// provide in component below
+		columns: [],
+		data: []
+	}
+};
+
+injectSortingArrowIcons({
+	ArrowUp: () => <div>↑</div>,
+	ArrowDown: () => <div>↓</div>
+});
+
+const CustomizedDataTable = () => {
+	const [rowSelection, setRowSelection] = useState<RowSelectionState>({});
+	const [sorting, setSorting] = useState<SortingState>([
+		{
+			id: "name",
+			desc: true
+		},
+		{
+			id: "age",
+			desc: false
+		}
+	]);
+
+	return (
+		<div>
+			<DataTable
+				columns={colsCustomized}
+				data={data}
+				getRowId={(x) => x.id.toString()}
+				rowSelection={rowSelection}
+				onRowSelectionChange={setRowSelection}
+				sorting={sorting}
+				onSortingChange={setSorting}
```

Comment:
i don't think sorting here actually works. any idea what's missing?

Reply:
There is also a `useSorting` hook involved, that is missing here I think :thinking:

Reply:
but that hook in pro is only for syncing the sorting state to the router/url. i don't think it's needed here because it's specific for our use case. or did i miss something and it has logic that makes the sorting work?

i checked it again, i see that the state gets set in the url and then taken in & applied. but can't we have the sorting work w/ local state first, instead of putting it in the url? at least for the example here

Reply:
Hm, I'm not sure tbh. The difference might be that in Pro were doing server-side sorting, while here you want to rely on client-only sorting https://tanstack.com/table/v8/docs/guide/sorting#client-side-vs-server-side-sorting